src/utils/tock.js

```diff
@@ -0,0 +1,117 @@
+const axios = require("axios");
+const { cache } = require("./cache");
+const { getSlackUsers } = require("./slack");
+
+const tockAPI = axios.create({
+  baseURL: process.env.TOCK_API,
+  headers: { Authorization: `Token ${process.env.TOCK_TOKEN}` },
+});
+
+const getFromTock = async (url) =>
+  cache(`tock fetch: ${url}`, 10, async () => {
+    const { data } = await tockAPI.get(url);
+    return data;
+  });
+
+/**
+ * Fetch a list of Tock users that are current 18F employees.
+ * @async
+ * @returns {Promise<Array<Object>>} A list of Tock users
+ */
+const getCurrent18FTockUsers = async () => {
+  // First get user data. This is what tells us whether users are current and
+  // are 18F employees. We'll use that to filter to just relevant users.
+  const userDataBody = await getFromTock(`/user_data.json`);
+
+  // Filter only current 18F employees. Only keep the user property. This
+  // is their username, and we'll use that to filter the later user list.
+  const userDataObjs = userDataBody
+    .filter((u) => u.is_active && u.current_employee && u.is_18f_employee)
+    .map((u) => u.user);
+
+  // Now get the list of users. This includes email addresses, which we can
+  // use to associate a user to a Slack account. However, this doesn't tell
+  // us whether they are currently an employee or with 18F, so we have to
+  // combine these two lists together.
```

Comment:
😭 (i think i sobbed about this on another PR)

Reply:
I KNOW RIGHT!